src/test/javascript/general.controller.test.js

```diff
@@ -303,208 +583,540 @@ describe("GeneralController", function () {
+            expect(arrayOfValidNames.toString()).toEqual("iamtst01,iamtst02,iamtst03,iamtst04,iamtst05,iamtst06,22222222,12345678,bogusname,fakename,_1234455,_iamtst01,_test_123-abc");
         });
-
     });
 
-    describe("validateAndAddUser", function () {
-        describe("user adds 'validUser', who is a valid user and is not in any list, to the Include list", function () {
-            const validUser = {
-                name: "Valid User",
-                username: "validUser",
-                uhUuid: "00000010",
-                firstName: "Valid",
-                lastName: "User"
-            };
-            beforeEach(function () {
-                scope.userToAdd = "validUser";
-                httpBackend.whenGET(BASE_URL + "currentUser")
-                    .respond(200);
-                httpBackend.whenGET(BASE_URL + "members/memberships/")
-                    .respond(200);
-                httpBackend.whenGET(BASE_URL + "owners/grouping/")
-                    .respond(200);
-                httpBackend.whenGET(BASE_URL + "members/" + validUser.username)
-                    .respond(200, validUser);
+    describe("getPersonProps", () => {
+        beforeEach(() => {
+            scope.personProps = "";
+        });
+        describe("get and modify person props", () => {
+            it("should return all the user's attributes", () => {
+                let propsArray = Object.keys(scope.groupingBasis[0]);
+                expect(propsArray).toEqual(["name", "username", "uhUuid", "firstName", "lastName"]);
+            });
+            it("should return a list with only the user's username, uhUuid, and name", () => {
+                spyOn(scope, "getPersonProps").and.callThrough();
+                let propsArray = Object.keys(scope.groupingBasis[0]);
+                scope.getPersonProps(propsArray);
+                expect(propsArray).not.toEqual(["name", "username", "uhUuid", "firstName", "lastName"]);
+                expect(scope.personProps).toEqual([["username"], ["uhUuid"], ["name"]]);
             });
+        });
+    });
 
-            it("user should be validated", function () {
-                spyOn(scope, "createConfirmAddModal").and.callThrough();
-                scope.validateAndAddUser(validUser.username, "Include");
-                httpBackend.expectGET(BASE_URL + "members/" + validUser.username)
-                    .respond(200, validUser);
+    describe("successfulAddHandler", () => {
+        let membersToAdd,
+            res,
+            invalidRes,
+            duplicateRes,
+            validPeople,
+            invalidMembersToAdd,
+            duplicateMembersToAdd,
+            duplicatePeople;
+        beforeEach(() => {
+            membersToAdd = "iamtst01,iamtst02,iamtst03,iamtst04";
+            invalidMembersToAdd = "invalidName01,invalidName02,invalidName03,invalidName04";
+            duplicateMembersToAdd = "iamtst01,iamtst01,iamtst03";
+            res = [
```

Comment:
Same thing with this array of objects, I think it would be beneficial to have a separate file which holds references to this data.